src/RadixDlt.NetworkGateway.PostgresIntegration/LedgerExtension/PostgresLedgerExtenderService.cs

```diff
@@ -947,13 +965,47 @@ private async Task<ExtendLedgerReport> ProcessTransactions(ReadWriteDbContext db
                     // TODO we'd love to see schemed JSON payload here and/or support for SBOR to schemed JSON in RET but this is not available yet; consider this entire section heavy WIP
                     foreach (var @event in committedTransaction.Receipt.Events)
                     {
-                        if (@event.Type.Emitter is not CoreModel.MethodEventEmitterIdentifier methodEventEmitter)
+                        if (@event.Type.Emitter is not CoreModel.MethodEventEmitterIdentifier methodEventEmitter ||
+                            methodEventEmitter.ObjectModuleId != CoreModel.ObjectModuleId.Main)
                         {
                             continue;
                         }
 
                         var eventEmitterEntity = referencedEntities.Get((EntityAddress)methodEventEmitter.Entity.EntityAddress);
 
+                        if (methodEventEmitter.Entity.EntityType == CoreModel.EntityType.GlobalValidator)
+                        {
+                            if (@event.Type.LocalTypeIndex.Index == eventTypeIdentifiers.GlobalValidator.ValidatorEmissionApplied)
+                            {
+                                var data = (JObject)@event.Data.ProgrammaticJson;
```

Comment:
BTW maybe we should decompile SBOR using RET.NET into something easier to ingest? Guess we might even need to define those event schemas... it's impossible to rely on `data[fields][4]` for anything long term. 

Ah... actually those events and their schema should already be available in package definition but until we get this "programmatic json with property names" we're doomed. Let's keep it this way... maybe just annotate it with TODO?